test/compilable/b19829.d

```diff
@@ -0,0 +1,5 @@
+static assert(!__traits(isSame, (i){ return i,x.value; }, a => a.value));
+static assert(!__traits(isSame, i => x[i].value, a => a.value));
+static assert(!__traits(isSame, i => [i].value, a => a.value));
+static assert(!__traits(isSame, i => new int, a => new int));
```

Comment:
these should be same, no?

Reply:
They probably should, yeah. I simply disallowed all behavior outside what was already covered.